qiskit_experiments/test/fake_service.py

```diff
@@ -0,0 +1,442 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Fake service class for tests."""
+
+from typing import Optional, List, Dict, Type, Any, Union, Tuple
+import json
+from datetime import datetime, timedelta
+import pandas as pd
+
+from qiskit_experiments.test.fake_backend import FakeBackend
+
+from qiskit_experiments.database_service import DatabaseServiceV1
+from qiskit_experiments.database_service.device_component import DeviceComponent
+from qiskit_experiments.database_service.exceptions import (
+    DbExperimentEntryExists,
+    DbExperimentEntryNotFound,
+)
+
+
+class FakeService(DatabaseServiceV1):
+    """
+    This extremely simple database is designated for testing and as a playground for developers.
+    It does not support multi-threading.
+    It is not guaranteed to perform well for a large amount of data.
+    It implements most of the methods of `DatabaseService`.
+    """
+
+    def __init__(self):
+        self.exps = pd.DataFrame(
+            columns=[
+                "experiment_type",
+                "backend_name",
+                "metadata",
+                "experiment_id",
+                "parent_id",
+                "job_ids",
+                "tags",
+                "notes",
+                "figure_names",
+            ]
+        )
+        self.results = pd.DataFrame(
+            columns=[
+                "experiment_id",
+                "result_data",
+                "result_type",
+                "device_components",
+                "tags",
+                "quality",
+                "verified",
+                "result_id",
+                "chisq",
+                "creation_datetime",
+                "service",
+            ]
+        )
+
+    def create_experiment(
+        self,
+        experiment_type: str,
+        backend_name: str,
+        metadata: Optional[Dict] = None,
+        experiment_id: Optional[str] = None,
+        parent_id: Optional[str] = None,
+        job_ids: Optional[List[str]] = None,
+        tags: Optional[List[str]] = None,
+        notes: Optional[str] = None,
+        json_encoder: Type[json.JSONEncoder] = json.JSONEncoder,
+        **kwargs: Any,
+    ) -> str:
```

Comment:
I found out that lint and sphinx don't require the doc strings in inheriting classes, and I thought this could be convenient, because duplicating these long doc strings from the parent class to the child class suffers from everything that's bad in duplication: it's exhausting to copy-paste all the time, and they'll soon diverge when one is modified: someone will modify one of them, and won't be aware that she has to modify the other one as well. 

On the other hand, when one reads the code, it's convenient to see the doc strings (I guess that it depends on the editor, maybe the editor can fetch them from the base class). In addition, in some methods I've added new exceptions of type `ValueError` (perhaps they won't be there anymore after the handling of the previous comment), so you'd expect it to appear in the doc string. But I don't want to copy such a long doc string only for the sake of adding a tiny "Raises ValueError".

What do you think?

Reply:
It's more than that - you are exposing the class; doesn't that mean that we expect sphinx to generate docs for it? If it can do this based on the parent class, great. Otherwise, we need docs, even if it means copy-pasting.

Your concern about divergence is correct, but on the other hand - divergence usually occurs when also actually changing the code, and since the parent is interface code, it means the inheriting classes need to be also modified anyway, so the docs should also be modified.

Reply:
>  If it can do this based on the parent class, great.

It does, I checked. Given this information, do you think I should copy nonetheless?

Reply:
No, don't copy.